msal/application.py

```diff
@@ -1560,6 +1676,59 @@ def acquire_token_interactive(
         self._validate_ssh_cert_input_data(kwargs.get("data", {}))
         claims = _merge_claims_challenge_and_capabilities(
             self._client_capabilities, claims_challenge)
+        if self._enable_broker:
+            try:
+                from .broker import _signin_interactively, _signin_silently
+            except RuntimeError:  # TODO: TBD
+                logger.debug("Broker is unavailable on this platform. Fallback to non-broker.")
+            else:
+                if extra_scopes_to_consent:  # TODO: Not supported in broker
+                    logger.warning(
+                        "Ignoring parameter extra_scopes_to_consent, "
+                        "which is not supported on current platform")
+                if "welcome_template" in kwargs:
+                    logger.debug(kwargs["welcome_template"])  # Experimental
+                authority = "https://{}/{}".format(
+                    self.authority.instance, self.authority.tenant)
+                validate_authority = (
+                    "no" if self.authority._validate_authority is False
+                        or self.authority.is_adfs or self.authority._is_b2c
+                    else None)
+                enable_msa_passthrough = self.client_id in [
+                    # Experimental: Automatically enable MSA-PT mode for known MSA-PT apps
+                    # More background of MSA-PT is available from this internal docs:
+                    # https://microsoft.sharepoint.com/:w:/t/Identity-DevEx/EatIUauX3c9Ctw1l7AQ6iM8B5CeBZxc58eoQCE0IuZ0VFw?e=tgc3jP&CID=39c853be-76ea-79d7-ee73-f1b2706ede05
+                    "04b07795-8ddb-461a-bbee-02f9e1bf7b46",  # Azure CLI
+                    "04f0c124-f2bc-4f59-8241-bf6df9866bbd",  # Visual Studio
+                    ] if enable_msa_passthrough is None else enable_msa_passthrough
+                # Call _signin_silently() and/or _signin_interactively()
+                if prompt == "none" or (not prompt and not login_hint):
+                    response = _signin_silently(
+                        authority, self.client_id, scopes,
+                        validateAuthority=validate_authority,
+                        claims=claims,
+                        max_age=max_age,
+                        enable_msa_pt=enable_msa_passthrough,
+                        **kwargs.get("data", {}))
+                    import pymsalruntime
+                    if prompt == "none" or response.get("_broker_status") not in (
+                            pymsalruntime.Response_Status.Status_AccountUnusable,
+                            pymsalruntime.Response_Status.Status_InteractionRequired,
+                            ):
+                        return self._process_broker_response(
+                            response, scopes, kwargs.get("data", {}))
+                response = _signin_interactively(
```

Comment:
>_signin_interactively [](http://example.com/codeflow?start=27&length=21)

Should we call _signin API instead?

Reply:
Indeed, that was how we did it here between [line 1704 to line 1730](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/415/files#diff-4d32d284fa6acf43894719eff11cb366e5969e0b61593bfd00c68a4cf9c441caR1704-R1730).